hw/scripts/stlink.sh

```diff
@@ -41,7 +41,11 @@ stlink_load () {
 
     echo "Downloading" $FILE_NAME "to" $FLASH_OFFSET
 
-    st-flash --reset write $FILE_NAME $FLASH_OFFSET
+    if which STM32_Programmer_CLI >/dev/null ; then
+      STM32_Programmer_CLI -c port=SWD -d $FILE_NAME $FLASH_OFFSET -rst
```

Comment:
It's a bit weird to force the use of the ST tool instead of using the open-source one provided by stlink (in a script called stlink.sh!). Have you considered using an environment variable, eg `STLINK_USE_STM32_PROGRAMMER_CLI=1` or something that people could enable the optional use?

Reply:
I though stlink is more like debugger name in that context, that's why I did not create separate script for this.
Even when ST environment is installed STM32_Programmer_CLI is buried inside some cryptic plugin folder:
/c/st/STM32CubeIDE_1.7.0/STM32CubeIDE/plugins/com.st.stm32cube.ide.mcu.externaltools.cubeprogrammer.win32_2.0.0.202105311346/tools/bin/

It will not be used by accident, user has to add this to PATH variable and when it's done user intention is obvious with no additional settings.

Reply:
> It will not be used by accident, user has to add this to PATH variable and when it's done user intention is obvious with no additional settings.

```
$ which STM32_Programmer_CLI
/home/utzig/STMicroelectronics/STM32Cube/STM32CubeProgrammer/bin/STM32_Programmer_CLI
```

Yes it will.